projects/hello_buttons_adcs/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include <stdint.h>
+
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress adc_pin_addr = { .port = GPIO_PORT_A, .pin = 6 };
+static GpioAddress button_addr = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioSettings adc_pin_settings = {
+  GPIO_DIR_IN,
+  GPIO_STATE_LOW,
+  GPIO_RES_NONE,
+  GPIO_ALTFN_ANALOG,
+};
+
+static InterruptSettings s_interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,       //
+  .priority = INTERRUPT_PRIORITY_NORMAL,  //
+};
+
+static AdcChannel adc_pin_channel = NUM_ADC_CHANNELS;
+
+static void prv_button_interrupt_handler(const GpioAddress *address, void *context) {
+  AdcChannel *adc_pin_channel = context;
+  uint16_t adc_pin_data = 0;
+
+  adc_read_converted(*adc_pin_channel, &adc_pin_data);
+  LOG_DEBUG("ADC data: %d\n", adc_pin_data);
+}
+
+void prv_register_interrupts(void) {
```

Comment:
Stylistic nit: this function should be marked `static` since it's a private function.